cpp/src/raft_runtime/neighbors/ivf_flat_serialize.cu

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sstream>
+#include <string>
+
+#include <raft/core/device_resources.hpp>
+#include <raft/neighbors/ivf_flat_serialize.cuh>
+#include <raft/neighbors/ivf_flat_types.hpp>
+#include <raft_runtime/neighbors/ivf_flat.hpp>
+
+namespace raft::runtime::neighbors::ivf_flat {
+
+#define RAFT_IVF_FLAT_SERIALIZE_INST(DTYPE)                                            \
+  void serialize_file(raft::resources const& handle,                                   \
+                      const std::string& filename,                                     \
+                      const raft::neighbors::ivf_flat::index<DTYPE, int64_t>& index)   \
+  {                                                                                    \
+    raft::neighbors::ivf_flat::serialize(handle, filename, index);                     \
+  };                                                                                   \
+                                                                                       \
+  void deserialize_file(raft::resources const& handle,                                 \
+                        const std::string& filename,                                   \
+                        raft::neighbors::ivf_flat::index<DTYPE, int64_t>* index)       \
+  {                                                                                    \
+    if (!index) { RAFT_FAIL("Invalid index pointer"); }                                \
+    *index = raft::neighbors::ivf_flat::deserialize<DTYPE, int64_t>(handle, filename); \
+  };                                                                                   \
+  void serialize(raft::resources const& handle,                                        \
```

Comment:
Note that this is not used by the current PR. But I think this can be used to fix https://github.com/rapidsai/cuml/issues/4743, that motivates to keep it in the PR.